drivers/serial/serial_esp32_usb.c

```diff
@@ -68,10 +68,14 @@ static void serial_esp32_usb_poll_out(const struct device *dev, unsigned char c)
 	ARG_UNUSED(dev);
 
 	/* Wait for space in FIFO */
+	const int64_t timeout =
+	k_uptime_ticks() +
+	(int64_t)k_us_to_ticks_ceil64(USBSERIAL_TIMEOUT_MAX_US);
```

Comment:
This timeout will be calculated for each call of the function. If no USB host is connected to the device (e.g. if powered by a simple USB power supply instead of a laptop), we will wait for the period of the timeout for each single character that should be sent out, wasting lots of MCU resources.

For this reason, the timeout must be stored statically and reset only when the fifo is ready again.

BTW: The indentation does not look correct.